component/src/main/java/com/celements/blog/article/Article.java

```diff
@@ -441,8 +488,124 @@ private XWikiContext getContext() {
   }
 
   @Override
-  public String toString() {
+  public @NotNull String toString() {
     return "Article [docRef=" + getDocumentReference() + "]";
   }
 
+  String getExtractPlainTextEncoded(String lang, boolean isViewtypeFull, int maxNumChars) {
+    String articleExtract = getExtract(lang, isViewtypeFull, maxNumChars);
+    try {
+      String plainExtract = new PlainTextCommand().convertHtmlToPlainText(articleExtract);
+      return plainExtract.replaceAll("\"", "&quot;");
+    } catch (ConvertToPlainTextException ctpte) {
+      LOGGER.error("Failed to convert article extract to plain text for article {}", articleDocRef,
+          ctpte);
+    }
+    return articleExtract;
+  }
+
+  List<ImageUrl> getArticleImagesBySizeAsc(String lang) {
+    List<ImageUrl> articleImages = getImageUrlExtractor().extractImagesSocialMediaUrlList(
+        getFullArticle(lang));
+    if (articleImages.isEmpty()) {
+      articleImages = getImageUrlExtractor().extractImagesSocialMediaUrlList(getExtract(lang, false,
+          getMaxNumChars()));
+    }
+    return articleImages;
+  }
+
+  public @NotNull String getExternalUrl() {
+    String fullName = getModelUtils().serializeRefLocal(articleDocRef);
+    try {
+      return context.getWiki().getExternalURL(fullName, "view", "ref=socialmedia", context);
+    } catch (XWikiException xwe) {
+      LOGGER.error("Exception while trying to get externalURL for doc {}", articleDocRef, xwe);
+    }
+    return context.getWiki().getURL(articleDocRef, "view", context);
+  }
+
+  /**
+   * Returns social media meta tags, respecting the configuration. If the tags are activate the
+   * calculated OpenGraph tags are added. Furthermore if a Twitter account is configured
+   * (e.g. @example), the Twitter tags are added. The Twitter card type is configurable, defaulting
+   * to "summary"
+   *
+   * @param language
+   */
+  public @NotNull List<MetaTag> getArticleSocialMediaTags(@NotNull String language) {
+    List<MetaTag> metaTags = new ArrayList<>();
+    if (getConfigurationSource().getProperty(BLOG_ARTICLE_SOCIAL_MEDIA_CONF_NAME, false)) {
+      List<ImageUrl> images = getArticleImagesBySizeAsc(language);
+      metaTags.addAll(addOpenGraphTags(images, language));
+      metaTags.addAll(addTwitterTags(images));
+    }
+    return metaTags;
+  }
+
+  List<MetaTag> addOpenGraphTags(List<ImageUrl> images, String language) {
```

Comment:
this method doesn't add tags, perhaps a better verb for naming would be "collect"?